lib/apiv2/felixconfig.go

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiv2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	KindFelixConfiguration     = "FelixConfiguration"
+	KindFelixConfigurationList = "FelixConfigurationList"
+)
+
+// Felix Configuration contains the configuration for Felix.
+type FelixConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the FelixConfiguration.
+	Spec FelixConfigurationSpec `json:"spec,omitempty"`
+}
+
+// TODO: Add validation on LogSeverityScreen, LogSeveritySys, LogSeverityFile, DatastoreType,
+// DefaultEndpointToHostAction, IptablesFilterAllowAction, IptablesMangleAllowAction,
+// ChainInsertModefor valid values.
+// FelixConfigurationSpec contains the values of the Felix configuration.
+type FelixConfigurationSpec struct {
+	UseInternalDataplaneDriver *bool  `json:"useInternalDataplaneDriver,omitempty"`
+	DataplaneDriver            string `json:"dataplaneDriver,omitempty"`
+
+	Ipv6Support    *bool `json:"ipv6Support,omitempty"`
+	IgnoreLooseRPF *bool `json:"ignoreLooseRPF,omitempty"`
+
+	// RouterefreshInterval is the period, in seconds, at which Felix re-checks the routes
+	// in the dataplane to ensure that no other process has accidentally broken Calico’s rules.
+	// Set to 0 to disable route refresh. [Default: 90]
+	RouteRefreshIntervalSecs *int `json:"routeRefreshIntervalSecs,omitempty" configname:"routeRefreshInterval"`
```

Comment:
The configname should have capital r:  "RouteRefreshInterval"

Reply:
sim. for other fields

Reply:
oops my bad.  Didn't catch that detail.